### PR TITLE
nodes: refactor process node handling to be simpler

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description
The logic in this file (node-managers-process-node) was hard to follow, so this is my attempt to simplify it.

I also modified the logic that reads the node from the db to use a pointer for target config so that we don't have to worry about always having a value there.

### :+1: Definition of Done
Tests still pass without modification and the code is easier to read.

### CODE REVIEW

File: components/nodemanager-service/pgdb/nodes.go [hunk 1]
```diff
@@ -153,22 +153,22 @@ var nodesSortFields = map[string]string{
...
+	ManagerIDs      json.RawMessage  `db:"manager_ids"`
+	Name            string           `db:"name"`
+	Platform        string           `db:"platform"`
+	PlatformVersion string           `db:"platform_version"`
+	Projects        json.RawMessage  `db:"projects"`
+	ReportID        string           `db:"report_id"`
+	State           string           `db:"source_state"`
+	Status          string           `db:"status"`
+	Tags            json.RawMessage  `db:"tags"`
+	TargetConfig    *json.RawMessage `db:"target_config"`
```
Comment (phiggins) [inline components/nodemanager-service/pgdb/nodes.go]: Could this change be avoided by checking for `if len(inNode.TargetConfig) > 0 {`?

I guess the deeper question is something like: do we need to differentiate between an empty TargetConfig vs nil?

Comment (vjeffrey) [reply components/nodemanager-service/pgdb/nodes.go]: we do - because without doing the pointer there you end up with a nasty sql error about scanning a nil value. it's because we didnt give it a default value when we first set this all up.

File: components/nodemanager-service/pgdb/node-managers-process-node.go [hunk 2]
```diff
@@ -55,98 +48,28 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
...
-	return nil
-}
+	lastContact := ptypes.TimestampString(node.LastContact)
 
-func updateNode(nodeUuid string, node *manager.NodeMetadata, lastContact string, db *DB, nodeState string) (sql.Result, error) {
-	updated, err := db.Exec(sqlUpdateNodeLastContact, nodeState, lastContact, node.JobUuid, node.Name, node.PlatformName, node.PlatformRelease, nodeUuid)
-	if err != nil {
-		return nil, errors.Wrapf(err, "updateNode unable to update node %s", node.Uuid)
+	if len(node.GetUuid()) == 0 {
+		node.Uuid = uuid.Must(uuid.NewV4()).String()
```
Comment (lancewf) [inline components/nodemanager-service/pgdb/node-managers-process-node.go]: We might want to return an error if we can not create the UUID instead of exiting the service.

Comment (vjeffrey) [reply components/nodemanager-service/pgdb/node-managers-process-node.go]: ahhh ya i forgot about that darn uuid must functionality, good catch - ya!
